{
  if (_connection == null) {
    return;
  }
  if (Debug.verboseOn())   Debug.logVerbose("SQLProcessor:commit() _manualTX=" + _manualTX,module);
  if (_manualTX) {
    try {
      _connection.commit();
      if (Debug.verboseOn())       Debug.logVerbose("SQLProcessor:commit() : called commit on connection",module);
    }
 catch (    SQLException sqle) {
      Debug.logError(sqle,"Error committing transaction: " + sqle.toString());
      try {
        rollback();
      }
 catch (      GenericDataSourceException rbsqle) {
        Debug.logError(rbsqle,"Got another error when trying to rollback after error committing transaction: " + sqle.toString());
      }
      throw new GenericDataSourceException("SQL Exception occurred on commit",sqle);
    }
  }
}
